amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -0,0 +1,219 @@
+                def vmOption = new Node(externalSystemNode, 'option', [name: "vmOptions", value: ""])
+                def systemDebugNode = new Node(configurationNode, 'GradleScriptDebugEnabled', null, true)
+                def methodNode = new Node(configurationNode, 'method', [v:"2"])
+
+                RunManagerNode.append(configurationNode)
+
+                //Save File
+                def writer = new FileWriter('./.idea/workspace.xml')
+
+                //Pretty print XML
+                groovy.xml.XmlUtil.serialize(xml, writer)
+            }
+        }
+
+        project.task('addAmplifyPushToWorkspace') {
+            //Open file
+            def xml = new XmlParser().parse('./.idea/workspace.xml')
+            def RunManagerNode = xml.component.find {
+                it.'@name' == 'RunManager'
+            } as Node
+            def configAmplifyPushCheck = RunManagerNode.children().find {
+                it.'@name' == 'amplifyPush'
+            } as Node
+
+            if (!configAmplifyPushCheck) {
+                // Nested nodes for amplifyPush run configuration
+                def configurationNode = new Node(null, 'configuration', [name: "amplifyPush", type:"GradleRunConfiguration", factoryName:"Gradle", nameIsGenerated:"true"])
+                def externalSystemNode = new Node(configurationNode, 'ExternalSystemSettings')
+                def executionOption = new Node(externalSystemNode, 'option', [name: "executionName"])
+                def projectPathOption = new Node(externalSystemNode, 'option', [name: "externalProjectPath", value: "\$PROJECT_DIR\$"])
+                def externalSystemIdOption = new Node(externalSystemNode, 'option', [name: "externalSystemIdString", value: "GRADLE"])
+                def scriptParametersOption = new Node(externalSystemNode, 'option', [name: "scriptParameters", value: ""])
+                def taskDescriptionsOption = new Node(externalSystemNode, 'option', [name: "taskDescriptions"])
+                def descriptionList = new Node(taskDescriptionsOption, 'list')
+                def taskNamesOption = new Node(externalSystemNode, 'option', [name: "taskNames"])
+                def nameList = new Node(taskNamesOption, 'list')
+                def amplifyPushOption = new Node(nameList, 'option', [value: "amplifyPush"])
+                def vmOption = new Node(externalSystemNode, 'option', [name: "vmOptions", value: ""])
+                def systemDebugNode = new Node(configurationNode, 'GradleScriptDebugEnabled', null, true)
+                def methodNode = new Node(configurationNode, 'method', [v:"2"])
+
+                RunManagerNode.append(configurationNode)
+
+                //Save File
+                def writer = new FileWriter('./.idea/workspace.xml')
+
+                //Pretty print XML
+                groovy.xml.XmlUtil.serialize(xml, writer)
+            }
+        }
+
+        project.createAmplifyApp.dependsOn('verifyNode')
```

Comment:
These are easier to keep track of if you put them after the task definitions. Like:
```
        project.task('createAmplifyApp') {
            // blah blah
        }
        project.createAmplifyApp.dependsOn('verifyNode')
```